Add support for remote file references with sysl modules

cuminandpaprika opened the pull request:
This PR replaces the `parse.NewParser().Parse(...)` call with `loader.LoadSyslModule(...)` so that references for remote imports are supported. Integration tests have also been added to ensure that catalog works correctly with remote modules.

The go.mod file added to the test package is necessary for sysl modules.

Note: Currently only works when the input file has no path e.g `sysl-catalog import.sysl` works but `sysl-catalog pkg/test/import.sysl` does not

joshcarp commented on main.go:
```diff
@@ -9,13 +9,13 @@ import (
 	"path"
 	"strings"
 
+	"github.com/anz-bank/sysl/pkg/loader"
 	"github.com/anz-bank/sysl/pkg/sysl"
 
 	"github.com/buger/goterm"
 
 	"github.com/anz-bank/sysl-catalog/pkg/catalog"
 	"github.com/anz-bank/sysl-catalog/pkg/watcher"
-	"github.com/anz-bank/sysl/pkg/parse"
 	"github.com/gohugoio/hugo/livereload"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/afero"
```
get rid of this file

joshcarp commented on pkg/test/go.mod:
```diff
@@ -0,0 +1,12 @@
+module github.com/anz-bank/sysl-catalog/pkg/test
+
+go 1.14
+
+require (
+	github.com/anz-bank/protoc-gen-sysl v0.0.8 // indirect
+	github.com/anz-bank/sysl v0.97.0
+	github.com/anz-bank/sysl-catalog v1.4.29
+	github.com/sirupsen/logrus v1.6.0
+	github.com/spf13/afero v1.2.2
+	github.com/stretchr/testify v1.5.1
+)
```
get rid of this file

cuminandpaprika replied:
It has to be there for sysl modules to work sadly :(

joshcarp replied:
why?

joshcarp replied:
you've given 0 explanation to any of the changes in here, so some elaboration would be nice

cuminandpaprika replied:
Sure, so sysl modules uses go modules under the hood to fetch remote sysl files. In order for that to work it needs a go.mod file (https://github.com/anz-bank/sysl/blob/0275e2741ea191f02e42dc1d3c8166a456cd7899/pkg/mod/module.go#L47)

Not exactly sure why the go.mod file is needed for go get to work, but I can ask @ChloePlanet for more details if you'd like (Or dig a little deeper myself).

joshcarp commented:
also the condition
 | Note: Currently only works when the input file has no path e.g sysl-catalog import.sysl works but sysl-catalog pkg/test/import.sysl does not
will break everything

cuminandpaprika commented:
> also the condition
> | Note: Currently only works when the input file has no path e.g sysl-catalog import.sysl works but sysl-catalog pkg/test/import.sysl does not
> will break everything

Ah really? I didn't think that'd be the primary way to call the sysl-catalog command, but retrospectively, it's probably mostly run from makefiles so makes sense that it'll need to support paths. 

I'll get that fixed up!

joshcarp commented on pkg/test/go.mod:
```diff
@@ -0,0 +1,12 @@
+module github.com/anz-bank/sysl-catalog/pkg/test
+
+go 1.14
+
+require (
+	github.com/anz-bank/protoc-gen-sysl v0.0.8 // indirect
+	github.com/anz-bank/sysl v0.97.0
+	github.com/anz-bank/sysl-catalog v1.4.29
+	github.com/sirupsen/logrus v1.6.0
+	github.com/spf13/afero v1.2.2
+	github.com/stretchr/testify v1.5.1
+)
```
get rid of this and the test file, 
Or test every dependency we've got, 
It doesn't make sense to me to have another test file within here, especially considering that we're only testing to see wether a local file can be loaded with sysl modules

cuminandpaprika replied:
Yup will delete!

joshcarp commented on main.go:
```diff
@@ -80,7 +80,7 @@ func main() {
 				}
 			}()
-			m, err = parse.NewParser().Parse(*input, fs)
+			m, _, err = loader.LoadSyslModule("", *input, fs, log)
```
You could just commit this, ya know

cuminandpaprika replied:
hahaha okay, will do! Just waiting for module version support to be merged in, then I'll bump the version and just add this change

cuminandpaprika committed:
Update sysl-catalog to use LoadSyslModule

Updates sysl-catalog to use LoadSyslModule instead of parse.NewParser().Parse so that remote files imported via //github.com/foo/bar are supported.
Bump sysl version to v0.98.0
